misk-jooq/src/main/kotlin/misk/jooq/JooqTransacter.kt

```diff
@@ -152,9 +168,73 @@ class JooqTransacter @JvmOverloads constructor(
     DataSourceType.VITESS_MYSQL -> SQLDialect.MYSQL
     DataSourceType.POSTGRESQL -> SQLDialect.POSTGRES
     DataSourceType.TIDB -> SQLDialect.MYSQL
+    DataSourceType.COCKROACHDB -> throw IllegalArgumentException("CockroachDB is not supported in JOOQ")
     else -> throw IllegalArgumentException("no SQLDialect for " + this.name)
   }
 
+  /**
+   * Determines if an exception should trigger a transaction retry.
+   * Adapted from Hibernate RealTransacter exception handling logic (without CockroachDB support).
+   */
+  private fun isRetryable(th: Throwable): Boolean {
+    return when (th) {
+      is SQLRecoverableException,
+      is SQLTransientException,
+      is OptimisticLockException -> true
+      is DataAccessException -> {
+        // Check if the underlying cause is retryable
+        val cause = th.cause
+        if (cause is SQLException) {
+          isMessageRetryable(cause) || isCauseRetryable(th)
+        } else {
+          // For DataAccessException without SQLException cause, always retry to match original behavior
+          true
+        }
+      }
+      is SQLException -> if (isMessageRetryable(th)) true else isCauseRetryable(th)
+      else -> isCauseRetryable(th)
+    }
+  }
+
+  private fun isMessageRetryable(th: SQLException) =
```

Comment:
Can we hoist these checks into `misk-jdbc` (or some other common library) and share between hibernate and jooq?